ibis/expr/rules.py

```diff
@@ -450,6 +452,62 @@ def python_literal(value, arg, **kwargs):
     return arg
 
 
+@functools.singledispatch
+def _make_hashable(value) -> Hashable:
```

Comment:
I feel like this should be done by `dt._normalize`

Reply:
I'd prefer to offload the data schema inference to more performant libraries, in this case directly to pandas since it is a mandatory dependency for now.

```python
@ibis.table.register(typing.Sequence)
def table_from_sequence(data, schema):
    # something like
    return pd.DataFrame(data, dtypes=ibis_schema_to_pandas(schema)).to_lists()  # something similar
```

Reply:
This is done now!